internal/commands/result.go

```diff
@@ -1040,6 +1040,40 @@ func setIsContainersEnabled(agent string, featureFlagsWrapper wrappers.FeatureFl
 	containerEngineCLIEnabled, _ := wrappers.GetSpecificFeatureFlag(featureFlagsWrapper, wrappers.ContainerEngineCLIEnabled)
 	wrappers.IsContainersEnabled = containerEngineCLIEnabled.Status && agentSupported
 }
+
+func filterScorecardResults(results *wrappers.ScanResultsCollection) []*wrappers.ScanResult {
+	var filteredResults []*wrappers.ScanResult
+	for _, result := range results.Results {
+		if result.Type != commonParams.SCSScorecardType {
+			filteredResults = append(filteredResults, result)
+		} else {
+			results.TotalCount--
+		}
+	}
+	return filteredResults
+}
+
+func filterScsResults(results *wrappers.ScanResultsCollection) []*wrappers.ScanResult {
```

Comment:
this function and filterScorecardResults function has almost the same logic. Convert this function and filterScorecardResults func to 1 function with an additional type param for filtering.